gobblin-data-management/src/main/java/org/apache/gobblin/data/management/source/LoopingDatasetFinderSource.java

```diff
@@ -79,59 +88,133 @@ public LoopingDatasetFinderSource(boolean drilldownIntoPartitions) {
     return Lists.newArrayList(getWorkunitStream(state).getMaterializedWorkUnitCollection());
   }
 
+  /**
+   * Collect all failed workunit in previous run into a list.
+   * <p>
+   * Workunits that exceed max retry limit will not be added into failedPreviousWorkUnits and rolled over into next
+   * quota-available execution.
+   */
+  private List<WorkUnit> getPreviousFailedWorkUnits(List<WorkUnitState> previousWorkUnitStates, int maxRetries) {
+    List<WorkUnit> failedPreviousWorkUnits = new ArrayList<>();
+    for (WorkUnitState workUnitState : previousWorkUnitStates) {
+      WorkUnitState.WorkingState workingState = workUnitState.getWorkingState();
+      if (workingState == WorkUnitState.WorkingState.FAILED || workingState == WorkUnitState.WorkingState.ROLLOVER) {
+        int currentRetryCount = 0;
+        if (workUnitState.contains(ConfigurationKeys.TASK_RETRIES_KEY)) {
+          currentRetryCount = workUnitState.getPropAsInt(ConfigurationKeys.TASK_RETRIES_KEY);
+        }
+        if (currentRetryCount < maxRetries && currentRetryCount > 0 ) {
```

Comment:
If TASK_RETRIES_KEY is not set, currentRetryCount will be 0 and the work unit will not be retried. We should remove the `currentRetryCount > 0` part and just update the log.info() message on the next line to not include the `currentRetryCount`.